README.md

```diff
@@ -136,6 +138,83 @@ Writing a new benchmark fixture would generally look like:
 
 The `benchmark_time` fixture provides a fairly simple example.
 
+## Investigating performance regressions
+
+It is not always obvious what the cause of a seeming performance regression is.
+It could be due to a new version of a direct or transitive dependency,
+and it could be due to a change in the Coiled platform.
+But often it is due to a change in `dask` or `distributed`.
+If you suspect that is the case, Coiled's `package_sync` feature combines well with the benchmarking infrastructure here and `git bisect`.
+
+The following is an example workflow which could be used to identify a specific commit in `dask` which introduced a performance regression.
+
+#### Create your software environment
+
+You should create a software environment which can run this test suite, but with an editable install of `dask`:
+```bash
+conda env create -n test-env --file ci/environment.yml
+conda activate test-env
+pip install .
+(cd <your-dask-dir> && pip install -e .)
+```
+
+#### Start bisecting
+
+Let's say the current `HEAD` of `dask` is known to be bad, and `$REF` is known to be good.
+In a separate terminal you can initialize a bisect workflow in your dask repository with
+
+```bash
+cd <your-dask-dir>
+git bisect start
+git bisect bad
+git bisect good $REF
+```
+
+#### Test for regressions
+
+Now that your editable install is bisecting, run a test or subset of tests which demonstrate the regression.
+Presume that `tests/benchmarks/test_parquet.py::test_write_wide_data` is such a test:
+
+```bash
+pytest tests/benchmarks/test_parquet.py::test_write_wide_data --benchmark
+```
+
+Once the test is done, it will have written a new entry to a local sqlite file `benchmark.db`.
+You will want to check whether that entry displays the regression.
+Exactly what that check will look like will depend on the test and the regression.
+You might have a script that builds a chart from `benchmark.db` similar to `dashboard.py`,
+or a script that performs some kind of statistical analysis.
+But let's assume a simpler case where you can recognize it from the `average_memory`.
+You can query that with
+
+```bash
+sqlite3 benchmark.db "select dask_version, average_memory from test_run where name = 'test_write_wide_data';"
+```
+
+If the last entry displays the regression, mark the commit in your dask terminal as `bad`:
+
+```bash
+git bisect bad
+```
+
+If the last entry doesn't display the regression, mark the commit in your dask terminal as `good`:
+
+```bash
+git bisect good
+```
+
+Proceed with this process until you have narrowed it down to a single commit.
+Congratulations! You've identified the source of your regression.
+
+#### Elaborations
```

Comment:
I wonder if this section should go on a follow up issue instead of having it in the README. I don't have a strong opinion about it, just a suggestion.